stdlib/SHA/src/sha2.jl

```diff
@@ -1,134 +1,162 @@
-function transform!(context::T) where {T<:Union{SHA2_224_CTX,SHA2_256_CTX}}
-    pbuf = buffer_pointer(context)
-    # Initialize registers with the previous intermediate values (our state)
-    a = context.state[1]
-    b = context.state[2]
-    c = context.state[3]
-    d = context.state[4]
-    e = context.state[5]
-    f = context.state[6]
-    g = context.state[7]
-    h = context.state[8]
-
-    # Run initial rounds
-    for j = 1:16
-        @inbounds begin
-            # We bitswap every input byte
-            v = bswap(unsafe_load(pbuf, j))
-            unsafe_store!(pbuf, v, j)
-
-            # Apply the SHA-256 compression function to update a..h
-            T1 = h + Sigma1_256(e) + Ch(e, f, g) + K256[j] + v
-            T2 = Sigma0_256(a) + Maj(a, b, c)
-            h = g
-            g = f
-            f = e
-            e = UInt32(d + T1)
-            d = c
-            c = b
-            b = a
-            a = UInt32(T1 + T2)
-        end
+macro R1_16(j, T)
+
+    ww = (:a, :b, :c, :d, :e, :f, :g, :h)
+
+    a = ww[((81 - j) % 8) + 1]
```

Comment:
You may want to verify that this generates fully optimized code without bounds checks and using a simple bitwise and operation for modular arithmetic.

Reply:
This is in the compile time part of the macro, so it doesn't matter.